src/components/AddUserX.tsx

```diff
@@ -0,0 +1,119 @@
+import React, { useState } from 'react';
+import { useMutation, gql } from '@apollo/client';
+import { Mutation_addUser } from '../schemaTypes';
+
+const AddUserX = (): JSX.Element => {
+  const [firstname, setFirstname] = useState('');
+  const [lastname, setLastname] = useState('');
+  const [email, setEmail] = useState('');
+  const [hash, setHash] = useState('');
+  const [role, setRole] = useState('');
+  const [position, setPosition] = useState('');
+
+  let input: any;
+
+  const ADD_USER = gql`
+    mutation Mutation($userInput: UserInput!) {
+      addUser(userInput: $userInput) {
+        _id
+        firstname
+        lastname
+        email
+        hash
+        role
+        position
+      }
+    }
+  `;
+  const [addUserFunction, { data, loading, error }] =
+    useMutation<Mutation_addUser>(ADD_USER);
+
+  console.log('hello');
+  return (
+    <div>
+      <form
+        onSubmit={(e) => {
```

Comment:
Put into a function outside of the return to make code cleaner and easier to read